### PR TITLE
Add goroutine safety doc on stream

### LINKED ISSUE #1298: Document the goroutine safety of the various interfaces
The Go documentation (https://godoc.org/google.golang.org/grpc) only refers to thread/goroutine safety for [`grpc.Codec`](https://godoc.org/google.golang.org/grpc#Codec). And that type is also the only mention of concurrency. A previous issue (https://github.com/grpc/grpc-go/issues/1015) touched on the need for understanding goroutine safety for `grpc.ClientConn`. Mailing list threads have asked about the safety of `grpc.Stream.SendMsg`. 

I am specifically interested in the safety of concurrent calls to `grpc.Stream.SendMsg` and `grpc.Stream.RecvMsg`. That is, can one goroutine call `SendMsg` while a different goroutine calls `RecvMsg` on the same stream? Perusing the code and testing indicate yes, but documentation of this would be appreciated.

### PR DESCRIPTION
**Pull request by menghanl**
fixes #1298

### CODE REVIEW

File: stream.go [hunk 1]
```diff
@@ -58,11 +58,17 @@ type Stream interface {
 	// side. On server side, it simply returns the error to the caller.
 	// SendMsg is called by generated code. Also Users can call SendMsg
 	// directly when it is really needed in their use cases.
+	// It's safe to have a goroutine calling SendMsg and another goroutine calling
+	// recvMsg on the same stream.
```
Comment (MakMukhi) [inline stream.go]: 'at' the same time

File: stream.go [hunk 2]
```diff
@@ -58,11 +58,17 @@ type Stream interface {
...
+	// But it is not safe to call SendMsg on same stream in different goroutines.
 	SendMsg(m interface{}) error
 	// RecvMsg blocks until it receives a message or the stream is
 	// done. On client side, it returns io.EOF when the stream is done. On
 	// any other error, it aborts the stream and returns an RPC status. On
 	// server side, it simply returns the error to the caller.
+	// It's safe to have a goroutine calling SendMsg and another goroutine calling
+	// recvMsg on the same stream.
```
Comment (MakMukhi) [inline stream.go]: same thing.

File: stream.go [hunk 3]
```diff
@@ -58,11 +58,17 @@ type Stream interface {
 	// side. On server side, it simply returns the error to the caller.
 	// SendMsg is called by generated code. Also Users can call SendMsg
 	// directly when it is really needed in their use cases.
+	// It's safe to have a goroutine calling SendMsg and another goroutine calling
+	// recvMsg on the same stream.
+	// But it is not safe to call SendMsg on same stream in different goroutines.
```
Comment (petermattis) [inline stream.go]: `s/on same/on the same/g`

File: stream.go [hunk 4]
```diff
@@ -58,11 +58,17 @@ type Stream interface {
...
 	SendMsg(m interface{}) error
 	// RecvMsg blocks until it receives a message or the stream is
 	// done. On client side, it returns io.EOF when the stream is done. On
 	// any other error, it aborts the stream and returns an RPC status. On
 	// server side, it simply returns the error to the caller.
+	// It's safe to have a goroutine calling SendMsg and another goroutine calling
+	// recvMsg on the same stream.
+	// But it is not safe to call RecvMsg on same stream in different goroutines.
```
Comment (petermattis) [inline stream.go]: Ditto (`on the same`).

**Comment by menghanl** [comment]
Thanks for the review. All fixed.

**Pull request merged**